l1-contracts/contracts/state-transition/StateTransitionManager.sol

```diff
@@ -96,6 +96,11 @@ contract StateTransitionManager is IStateTransitionManager, ReentrancyGuard, Own
         return IBridgehub(BRIDGE_HUB).getHyperchain(_chainId);
     }
 
+    /// @notice return the state transition chain contract for a chainId
+    function getHyperchainLegacy(uint256 _chainId) public view returns (address chainAddress) {
```

Comment:
please add a comment that it should not be relied upon (not 100% sure that `legacy` part in the name sends the message that it will be removed soon)